lib/Immutable.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+const util = require('util');
+
+/**
+ * Error for Immutable variables
+ *
+ * @extends Error
+ * @class Errors.Immutable
+ * */
+class Immutable extends Error{
+    /**
+ * @example
+
+      let piKindOf = 22/7;
+
+      function getItRight(){
+          if(typeof piKindOf != 'undefined'){
+              const err = new Errors.Immutable;
+              err.setMessage(
+                  'piKindOf',
+                  'global'
+              );
+              throw err;
+          }
+          piKindOf = Math.PI;
+      }
+
+      getItRight();
+
+
+ * @example
+ *
+ * ```sh
+ *
+ /git/node-error-classes/example/immutable.js:14
+     throw err;
+     ^
+
+Immutable: 'piKindOf' has been defined and cannot be changed on the scope of : 'global'.
+     Variable names here must be unique
+
+ * ```
+ *
+ * @method Errors.Immutable.setMessage
+ * @param {String} varaibleName Name of variable
+ * @param {Any} scope     Scope of varaible
+ * @returns {String}    compiled error message
+ */
+    setMessage(varaibleName, scope){
+        this.name = 'Immutable';
+        this.message = `${util.inspect(varaibleName, {colors: true})} has been defined and cannot be changed on the scope of : ${util.inspect(scope, {colors: true})}.
+        Variable names here must be unique`;
```

Comment:
The scope should be util inspected as an object. Please follow the other examples. I should be able to pass in an object and have it logged at a depth of 5.

If I pass {a:1,b:2} to this error as the scope or "this". I used depth:5 in all circumstances in other errors.

You put it to null last commit so I told you id didn't need to be there, but conversely if you are logging a string at depth 5 it still looks like a string.

It would allow a user to pass either a string or an object here. plz always use depth:5 as in the other errors.